### PR TITLE
LRA-563-lsa-ride-share-unit-admin-notification-for-reservation

### PR DESCRIPTION
**Pull request by rsmoke**
Create a mailer that will send a message to the UNIT ADMIN from submitter (current-user) after a reservation request is submitted.

EXPECTATIONS:

STUDENT USER

A new car reservation is created.

Upon clicking the [Create Reservation] button an email is sent to the UNIT ADMIN for the unit that the reservation was created in.

SUBJECT: New reservation for program: #{@reservation.program.title}

BODY: <%= link_to "#{@reservation.display_name}", reservation_url(@reservation)” %> 
Details:
PROGRAM: <%= link_to <Program_id>, program_url %>
SITE: <%= link_to <Site_id>, site_url %> 
CAR #: <%= link_to <Car_id>, car_url %>
DATE: <Start_Datetime> - <End_Datetime>

Thank you,

ADMIN USER:

Rita is checking to see if a message should be sent out if an admin user creates the reservation. The question is if there are more than one admin then the message would notify the admin group of a unit.

### CODE REVIEW

**Comment by britaumich** [comment]
@rsmoke  After I created a reservation (clicked Create Reservation button) I was redirected to the file:///Users/brita/lsa_ride_share/tmp/letter_opener/1686867469_796996_82c8ceb/rich.html URL with an email text (see the image below). This is not what is supposed to happen. The app should go to the add drivers page.
Screenshot 2023-06-15 at 6 20 57 PM

**Comment by rsmoke** [comment]
I believe you're seeing this because you weren't running the letter opener app. Did you bundle install to be sure that the letter opener web gem was in place?

File: app/controllers/reservations_controller.rb [hunk 1]
```diff
@@ -170,6 +171,7 @@ def update
     @reservation.number_of_people_on_trip = params[:number_of_people_on_trip]
     respond_to do |format|
       if @reservation.update(reservation_params)
+        # ReservationMailer.with(reservation: @reservation).car_reservation_created.deliver_now
```
Comment (anantaUM) [inline app/controllers/reservations_controller.rb:174]: Please remove this commented code.

File: app/mailers/reservation_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class ReservationMailer < ApplicationMailer
+  def car_reservation_created
+    @reservation = params[:reservation]
+    @reserved_by_name = User.find(@reservation.reserved_by).display_name
+    mail(to: @reservation.program.unit.unit_preferences.find_by(name: "notification_email").value, subject: "New reservation for program: #{@reservation.program.title}" )
```
Comment (anantaUM) [inline app/mailers/reservation_mailer.rb:5]: _@reservation.program.unit.unit_preferences.find_by(name: "notification_email").value_

There is a chain of function calls happening in the above code. Is there a possibility that any one of them might return null or blank value ? Do we need any additional validation checks ?

**Review by anantaUM — COMMENTED** [review]
Added comments.

**Comment by britaumich** [comment]
@rsmoke bundle did the trick, though I installed the letter opener when you just added it to the app. I have 'Hello, you've got mail!' from May 10th. I'll look at the request's code a little bit later.

File: app/mailers/reservation_mailer.rb (hunk 2 above)
Comment (rsmoke) [reply app/mailers/reservation_mailer.rb:5]: Good point. I will add default values in the parent template to be sure there is not a nil value.

Comment (rsmoke) [reply app/mailers/reservation_mailer.rb:5]: defaults have been added

**Review by rsmoke — COMMENTED** [review]
updated from comments

File: app/mailers/reservation_mailer.rb (hunk 2 above)
Comment (britaumich) [reply app/mailers/reservation_mailer.rb:5]: @rsmoke if @reservation.program.unit.unit_preferences.find_by(name: "notification_email").value is nil or empty mail command fails with "SMTP To address may not be blank". How the mailer can use the default to: value?

File: app/controllers/reservations_controller.rb (hunk 1 above)
Comment (rsmoke) [reply app/controllers/reservations_controller.rb]: removed code

**Review by britaumich — APPROVED** [review]

**Pull request merged**